day7/part_1.py

```diff
@@ -0,0 +1,93 @@
+"""
+For solving advent of code problem 7 part 1
+"""
+from collections import Counter
+import numpy as np
+from scipy.stats import rankdata
+
+
+def combo_score(card: str):
+    """Returns the combination score of a hand.
+    
+    The score is the concatenated count of the 2 most common cards.
+    That is:
+    combo_score of five of a kind is 50
+    combo_score of four of a kind is 41
+    combo_score of full house is 32
+    combo_score of three of a kind is 31
+    combo_score of two pair is 22
+    combo_score of pair is 21
+    combo_score of high card is 11
+    
+    Example:
+    >>>combo_score('33556')
+    22
+    This is because we have two pair.
+    """
+    char_count = Counter(card)
+    most_common_chars = char_count.most_common(2)
+    most_common_count = most_common_chars[0][1] if most_common_chars else 0
+    second_most_common_count = most_common_chars[1][1] if len(most_common_chars) >= 2 else 0
+
+    return int(most_common_count) * 10 + int(second_most_common_count)
+
+def tie_break_score(card: str):
+    """Returns the tie break score of a hand.
+    
+    The tie break score is calculated as follows:
+    value of first card times 100,000,000 plus
+    value of second card times 1,000,000 plus
+    value of third card times 10,000 plus
+    value of fourth card times 100 plus
+    value of fifth card times 1
+    
+    Example:
+    >>>tie_break_score('AT72Q')
+    1410070212
+    """
+    values = {'A':14, 'K':13, 'Q':12, 'J':11, 'T':10}
+    power = 4
+    score = 0
+    for char in card:
+        if char.isdigit():
+            value = int(char)
+        else:
+            value = values[char]
+        score += value * 10 ** (2*power)
```

Comment:
I agree with wasted space. At first, I wanted to encode in the hexadecimal because we have conveniently have 14 cards. However, I decided to go with this way to make the solution more visual.